sway/commands/input/scroll_factor.c

```diff
@@ -0,0 +1,32 @@
+#include <errno.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include "sway/config.h"
+#include "sway/commands.h"
+#include "sway/input/input-manager.h"
+#include "util.h"
+
+struct cmd_results *input_cmd_scroll_factor(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "scroll_factor", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+	struct input_config *ic = config->handler_context.input_config;
+	if (!ic) {
+		return cmd_results_new(CMD_FAILURE,
+			"scroll_factor", "No input device defined.");
+	}
+
+	float scroll_factor = parse_float(argv[0]);
+	if (scroll_factor == NAN) {
```

Comment:
Sorry. I missed this before. You cannot compare a floating point value with `NAN` to determine whether the value is `NAN` since `NAN != NAN`. You can either use `isnan(scroll_factor)` or `scroll_factor != scroll_factor`.